File: DynamicProgramming/FibonacciSequence/dp.c
// Fibonacci Series using Dynamic Programming
#include <stdio.h>
#include <time.h>
 
long long int fib(int n)
{
    /* Declare an array to store Fibonacci numbers. */
    long long int f[n+2]; // 1 extra to handle case, n = 0
    int i;
 
    /* 0th and 1st number of the series are 0 and 1*/
    f[0] = 0;
    f[1] = 1;
 
    for (i = 2; i <= n; i++) {
        /* Add the previous 2 numbers in the series
           and store it */
        f[i] = f[i - 1] + f[i - 2];
    }
 
    return f[n];
}
 
int main()
{
    clock_t start = clock();

    int n = 48;
    printf("%lld\n", fib(n));

    clock_t end = clock();
    double elapsed_time = ((double) (end - start)) / CLOCKS_PER_SEC * 1000.0;
    printf("elapsed time: %f milliseconds\n", elapsed_time);

    // getchar();
    return 0;
}

File: DynamicProgramming/FibonacciSequence/dp.py
import time
# Fibonacci Series using Dynamic Programming
def fibonacci(n):
 
    # Taking 1st two fibonacci numbers as 0 and 1
    f = [0, 1]
 
    for i in range(2, n+1):
        f.append(f[i-1] + f[i-2])
    return f[n]
 
start_time = time.time() 
print("fibonacci 48: ", fibonacci(48))
end_time = time.time()
elapsed_time = (end_time - start_time) *1000
print("elapsed time: ", elapsed_time, "seconds")
